src/source.c

```diff
@@ -1881,13 +1895,15 @@ _dispatch_timers_program2(uint64_t nows[], _dispatch_kevent_qos_s *ke,
 		_dispatch_trace_next_timer_set(
 				TAILQ_FIRST(&_dispatch_kevent_timer[tidx].dk_sources), qos);
 		_dispatch_trace_next_timer_program(delay, qos);
-		delay += _dispatch_source_timer_now(nows, DISPATCH_TIMER_KIND_WALL);
+	        nowtime =_dispatch_source_timer_now(nows, DISPATCH_TIMER_KIND_WALL);
+	        adjust_delay(delay,nowtime);
```

Comment:
should be removed, and libkqueue updated to do conversions from ns to ms internally instead.